src/test/kotlin/org/rust/ide/annotator/RsExpressionAnnotatorTest.kt

```diff
@@ -98,4 +98,18 @@ class RsExpressionAnnotatorTest : RsAnnotatorTestBase() {
             let _ = U <error descr="Union expressions should have exactly one field">{ a: 92, b: 92.0}</error>;
         }
     """)
+
+    fun testReassignImmutableFromBinding() = checkErrors("""
+        fn main() {
+            let x = 5;
+            <error descr="Reassigning an immutable variable [E0384]">x = 3</error>;
+        }
+    """)
+
+    fun testReassignMutableFromBinding() = checkErrors("""
+        fn main() {
+            let mut x = 5;
+            x = 3;
+        }
+    """)
```

Comment:
I usually add error numbers to the tests, like `testE0384_ReassigningImmutableVariable()` and put them in the numerical order. Maybe we could stick to such rule to keep them manageable as their number grows.